core/templates/services/math-interactions.service.ts

```diff
@@ -171,10 +172,12 @@ export class MathInteractionsService {
         'Your answer contains an invalid term: ' + invalidIntegers[0]);
       return false;
     }
-    let invalidMultiTerms = expressionString.match(/([a-zA-Zα-ωΑ-Ω]\d+)/g);
+    let invalidMultiTerms = expressionString.match(/([a-zA-Z]+\d+)/g);
     if (invalidMultiTerms !== null) {
+      let firstNumberIndex = invalidMultiTerms[0].search(/\d/);
```

Comment:
What if the numeric part is, say, 34? (i.e. more than one digit)

Reply:
We just need the index to split the term by. We don't need to search for the entire numeric part.
As added in the unit test for this, if the term is `omega34`, `firstNumberIndex` would be index of `3`, and that is the point we want to split and swap.